app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,58 @@
+class VendorsController < ApplicationController
+  def index
+    @all_vendors = Vendor.all.order(name: :asc)
+    render :index
+  end
+
+  def show
+    @vendor = Vendor.find(params[:id])
+    @all_products = Product.all
```

Comment:
Here is where you could get just @vendor.products since that association is already made!